examples/preapprovals/components/segmented-control.tsx

```diff
@@ -1,66 +1,84 @@
+  { label: string }
+>(function SegmentedControlOption({ label, ...inputProps }, ref) {
   const theme = useTheme();
+  const id = useId();
 
   return (
-    <Box height="medium" display="flex">
-      {options.map(option => (
-        <Box
-          key={option.value}
-          className={css({
-            flexBasis: 0,
-            flexGrow: 1,
-            position: 'relative',
-            ':first-child > label': {
-              borderTopLeftRadius: theme.border.radius[borderRadius],
-              borderBottomLeftRadius: theme.border.radius[borderRadius],
-            },
-            ':last-child > label': {
-              borderTopRightRadius: theme.border.radius[borderRadius],
-              borderBottomRightRadius: theme.border.radius[borderRadius],
+    <Box
+      flex={1}
+      position="relative"
+      className={css({
+        ':first-child > label': {
+          borderTopLeftRadius: theme.border.radius[borderRadius],
+          borderBottomLeftRadius: theme.border.radius[borderRadius],
+        },
+        ':last-child > label': {
+          borderTopRightRadius: theme.border.radius[borderRadius],
+          borderBottomRightRadius: theme.border.radius[borderRadius],
+        },
+      })}
+    >
+      <VisuallyHidden
+        {...inputProps}
+        as="input"
+        type="radio"
+        id={id}
+        ref={ref}
+      />
+      <Box
+        as="label"
+        htmlFor={id}
+        background="surface"
+        cursor="pointer"
+        display="flex"
+        alignItems="center"
+        justifyContent="center"
+        height="full"
+        userSelect="none"
+        className={css({
+          ':hover': {
+            background: theme.backgroundInteractions.primaryLowHover,
+          },
+          'input:checked + &': {
+            background: theme.color.background.primary,
+            '*': {
+              color: theme.color.foreground.neutralInverted,
             },
-          })}
-        >
-          <Box as="input" type="radio" id={option.value} name={name} />
-          <Box
-            as="label"
-            htmlFor={option.value}
-            background="surface"
-            border="field"
-            cursor="pointer"
-            className={css({
-              position: 'absolute',
-              inset: 0,
-              ':hover': {
-                borderColor: theme.border.color.primaryHover,
-              },
-              'input:checked + &': {
-                backgroundColor: theme.color.background.primary,
-                '*': {
-                  color: theme.color.foreground.neutralInverted,
-                },
-              },
-              display: 'flex',
-              alignItems: 'center',
-              justifyContent: 'center',
-            })}
-          >
-            <Text>{option.label}</Text>
-          </Box>
-        </Box>
-      ))}
+          },
+          'input:active + &': {
+            background: theme.backgroundInteractions.primaryLowActive,
+          },
```

Comment:
I just realised we should probably add focus styles.

Something like this should work:

```
// Import the focus styles from '@spark-web/a11y'
const focusStyles = useFocusRing();
```

And apply them when focused (along with position: relative and a z-index so the focus ring will be above the sibling radio inputs:

```suggestion
          'input:active + &': {
            background: theme.backgroundInteractions.primaryLowActive,
          },
          'input:focus + &': {
            ...focusStyles,
            position: 'relative',
            zIndex: theme.elevation.sticky,
          },
```

Reply:
done ✅